sympy/printing/smtlib.py

```diff
@@ -248,23 +247,24 @@ def smtlib_code(
         A list of lists of ``str`` and/or expressions to convert into SMTLib and prefix to the output.
     suffix_expressions: list, optional
         A list of lists of ``str`` and/or expressions to convert into SMTLib and postfix to the output.
-    log_warn: lambda function, optional
-        A function to record all warnings during potentially risky operations.
-        Soundness is a core value in SMT solving, so it is good to log all assumptions made.
-        If not given, builtins ``print`` will be used.
 
     Examples
     ========
+    >>> import warnings
+    >>> warnings.showwarning = (lambda m, *_: print(repr(m)))
```

Comment:
Does doctest support warnings? Changing `showwarnings` like this is problematic because it changes the warnings module globally, meaning it will still be in effect after the doctests run.

Reply:
It doesn't seem like there's a great way to do this: https://stackoverflow.com/questions/2418570/testing-warnings-with-doctest

If you think it's sufficient, I can just make sure to turn `showwarnings` back when I'm done using it?

Reply:
I guess this is fine for now. My guess is we can pipe stderr to stdout [here](https://github.com/sympy/sympy/blob/8a5d62234c0713fa75c236193295cb21afb3ef8d/sympy/testing/runtests.py#L1815) to make warnings work in doctests, but I haven't tried it so I can't say for sure how straightforward it will be.